test/gateway/eip20gateway/helpers/eip20gateway.js

```diff
@@ -0,0 +1,209 @@
+        gasLimit,
+        nonce,
+        hashLock,
+        signature,
+        txOptions
+      );
+
+      assert.equal(
+        response.receipt.status,
+        1,
+        "Receipt status is unsuccessful"
+      );
+      let eventData = response.logs;
+      utils.validateEvents(eventData, expectedResults.events);
+    }
+
+    // Get the final balances for all the addresses involved.
+    let stakerTokenBalance = await oThis.token.balanceOf.call(staker),
+      gatewayTokenBalance = await oThis.token.balanceOf.call(
+        oThis.gateway.address
+      ),
+      facilitatorTokenBalance = await oThis.token.balanceOf.call(
+        txOptions.from
+      );
+
+    // Get the final balances of base token for all the addresses involved.
+    let stakerBaseTokenBalance = await oThis.baseToken.balanceOf.call(staker),
+      gatewayBaseTokenBalance = await oThis.baseToken.balanceOf.call(
+        oThis.gateway.address
+      ),
+      facilitatorBaseTokenBalance = await oThis.baseToken.balanceOf.call(
+        txOptions.from
+      );
+
+    // Assert the balances
+    if (resultType == utils.ResultType.FAIL) {
+
+      assert(
+        initialStakerTokenBalance.eq(stakerTokenBalance),
+        "Staker balance must be unchanged"
+      );
+
+      assert(
+        initialGatewayTokenBalance.eq(gatewayTokenBalance),
+        "Gateway balance must be unchanged"
+      );
+
+      assert(
+        initialFacilitatorTokenBalance.eq(facilitatorTokenBalance),
+        "Facilitator balance must be unchanged"
+      );
+
+      assert(
+        initialStakerBaseTokenBalance.eq(stakerBaseTokenBalance),
+        "Staker balance for base token must be unchanged"
+      );
+
+      assert(
+        initialGatewayBaseTokenBalance.eq(gatewayBaseTokenBalance),
+        "Gateway balance for base token must be unchanged"
+      );
+
+      assert(
+        initialFacilitatorBaseTokenBalance.eq(facilitatorBaseTokenBalance),
+        "Facilitator balance for base token must be unchanged"
+      );
+
+    } else {
+
+      assert(
+        stakerTokenBalance.eq(initialStakerTokenBalance.sub(amount)),
+        `Staker balance must decrease by ${amount}`
+      );
+
+      assert(
+        initialGatewayTokenBalance.add(amount).eq(gatewayTokenBalance),
+        `Gateway balance must increase by ${amount}`
+      );
+
+      assert(
+        initialFacilitatorTokenBalance.eq(facilitatorTokenBalance),
+        "Facilitator balance must be unchanged"
+      );
+
+      assert(
+        initialStakerBaseTokenBalance.eq(stakerBaseTokenBalance),
+        "Staker balance for base token must be unchanged"
+      );
+
+      assert(
+        initialGatewayBaseTokenBalance.add(bounty).eq(gatewayBaseTokenBalance),
+        `Gateway balance must create by ${bounty.toString(10)}`
+      );
+
+      assert(
+        initialFacilitatorBaseTokenBalance.sub(bounty).eq(facilitatorBaseTokenBalance),
+        `Facilitator balance must decrease by ${bounty}`
+      );
+
+    }
+  }
```

Comment:
This function is extremely long. Any chance it can be split into smaller, more understandable functions?

Reply:
Broken into 2 functions!!!  😅